src/uthash.h

```diff
@@ -168,33 +192,42 @@ do {
   if (!(head)->hh.tbl) {                                                         \
-    uthash_fatal("out of memory");                                               \
-  }                                                                              \
-  uthash_bzero((head)->hh.tbl, sizeof(UT_hash_table));                           \
-  (head)->hh.tbl->tail = &((head)->hh);                                          \
-  (head)->hh.tbl->num_buckets = HASH_INITIAL_NUM_BUCKETS;                        \
-  (head)->hh.tbl->log2_num_buckets = HASH_INITIAL_NUM_BUCKETS_LOG2;              \
-  (head)->hh.tbl->hho = (char*)(&(head)->hh) - (char*)(head);                    \
-  (head)->hh.tbl->buckets = (UT_hash_bucket*)uthash_malloc(                      \
-      HASH_INITIAL_NUM_BUCKETS * sizeof(struct UT_hash_bucket));                 \
-  if (!(head)->hh.tbl->buckets) {                                                \
-    uthash_fatal("out of memory");                                               \
+    HASH_MEM_FAILED(mem_ok);                                                     \
+  } else {                                                                       \
+    uthash_bzero((head)->hh.tbl, sizeof(UT_hash_table));                         \
+    (head)->hh.tbl->tail = &((head)->hh);                                        \
+    (head)->hh.tbl->num_buckets = HASH_INITIAL_NUM_BUCKETS;                      \
+    (head)->hh.tbl->log2_num_buckets = HASH_INITIAL_NUM_BUCKETS_LOG2;            \
+    (head)->hh.tbl->hho = (char*)(&(head)->hh) - (char*)(head);                  \
+    (head)->hh.tbl->buckets = (UT_hash_bucket*)uthash_malloc(                    \
+        HASH_INITIAL_NUM_BUCKETS * sizeof(struct UT_hash_bucket));               \
+    if (!(head)->hh.tbl->buckets) {                                              \
+      HASH_MEM_FAILED(mem_ok);                                                   \
+      uthash_free((head)->hh.tbl, sizeof(UT_hash_table));                        \
+    } else {                                                                     \
+      uthash_bzero((head)->hh.tbl->buckets,                                      \
+          HASH_INITIAL_NUM_BUCKETS * sizeof(struct UT_hash_bucket));             \
+      HASH_BLOOM_MAKE((head)->hh.tbl, mem_ok);                                   \
+      if (HASH_NOMEM_OK && !(mem_ok)) {                                          \
+        uthash_free((head)->hh.tbl->buckets,                                     \
+            HASH_INITIAL_NUM_BUCKETS*sizeof(struct UT_hash_bucket));             \
+        uthash_free((head)->hh.tbl, sizeof(UT_hash_table));                      \
+      } else {                                                                   \
+        (head)->hh.tbl->signature = HASH_SIGNATURE;                              \
+      }                                                                          \
```

Comment:
This (control flow) was exactly the reason I didn't do this before. I figured the `HASH_SIGNATURE` assignment was deliberately done at the end to make sure the hash didn't complete initialization, but had that field set. Anyway, fixed.